kernel/aster-nix/src/net/socket/ip/stream/mod.rs

```diff
@@ -376,7 +379,32 @@ impl FileLike for StreamSocket {
         self.pollee.poll(mask, poller)
     }
 
+    fn ioctl(&self, cmd: IoctlCmd, arg: usize) -> Result<i32> {
```

Comment:
I think you can define and implement the `ioctl` method in the `Socket` trait, so all the structs which implement `Socket` can handle the `ioctl`.